agenta-backend/agenta_backend/services/evaluation_service.py

```diff
@@ -0,0 +1,169 @@
+            **extend_with_evaluation(newEvaluationData.evaluation_type),
+            **extend_with_correct_answer(newEvaluationData.evaluation_type, datum)
+        }
+
+        await evaluation_scenarios.insert_one(evaluation_scenario)
+
+    evaluation["id"] = str(newEvaluation.inserted_id)
+    return evaluation
+
+
+async def update_evaluation_scenario(
+    evaluation_scenario_id: str,
+    evaluation_scenario_data: EvaluationScenarioUpdate,
+    evaluation_type: EvaluationType
+) -> Dict:
+    evaluation_scenario_dict = evaluation_scenario_data.dict()
+    evaluation_scenario_dict["updated_at"] = datetime.utcnow()
+
+    new_evaluation_set = {"outputs": evaluation_scenario_dict["outputs"]}
+
+    if (
+        evaluation_type == EvaluationType.auto_exact_match
+        or evaluation_type == EvaluationType.auto_similarity_match
+    ):
+        new_evaluation_set["score"] = evaluation_scenario_dict["score"]
+    elif evaluation_type == EvaluationType.human_a_b_testing:
+        new_evaluation_set["vote"] = evaluation_scenario_dict["vote"]
+    elif evaluation_type == EvaluationType.auto_ai_critique:
+        current_evaluation_scenario = await evaluation_scenarios.find_one(
+            {"_id": ObjectId(evaluation_scenario_id)}
+        )
+        current_evaluation = await evaluations.find_one({"_id": ObjectId(current_evaluation_scenario["evaluation_id"])})
+
+        evaluation = evaluate_with_ai_critique(
+            llm_app_prompt_template=current_evaluation["llm_app_prompt_template"],
+            llmm_app_inputs=current_evaluation_scenario["inputs"],
+            correct_answer=current_evaluation_scenario["correct_answer"],
+            app_variant_output=new_evaluation_set["outputs"][0]["variant_output"],
+            evaluation_prompt_template=evaluation_scenario_dict["evaluation_prompt_template"],
+            open_ai_key=evaluation_scenario_dict["open_ai_key"],
+        )
+
+        new_evaluation_set["evaluation"] = evaluation
+
+    result = await evaluation_scenarios.update_one(
+        {"_id": ObjectId(evaluation_scenario_id)}, {"$set": new_evaluation_set}
+    )
+    if result.acknowledged:
+        evaluation_scenario = await evaluation_scenarios.find_one({"_id": ObjectId(evaluation_scenario_id)})
+
+        if evaluation_scenario:
+            evaluation_scenario["id"] = str(evaluation_scenario["_id"])
+            del evaluation_scenario["_id"]
+            return evaluation_scenario
+
+    raise UpdateEvaluationScenarioError("Failed to create evaluation_scenario")
+
+
+def evaluate_with_ai_critique(
+        llm_app_prompt_template: str,
+        llmm_app_inputs: dict,
+        correct_answer: str,
+        app_variant_output: str,
+        evaluation_prompt_template: str,
+        open_ai_key: str,
+        temperature: float = 0.9) -> str:
+
+    llm = OpenAI(openai_api_key=open_ai_key, temperature=temperature)
+
+    input_variables = ["app_variant_output", "llm_app_prompt_template", "correct_answer"]
```

Comment:
This would be more pythonic
```suggestion
    input_variables = ["app_variant_output", "llm_app_prompt_template", "correct_answer"] + [_['input_name'] for _ in llmm_app_inputs]
```